src/sota_tools/authenticate_test.cc

```diff
@@ -63,9 +62,14 @@ TEST(authenticate, good_cert_noauth_zip) {
 }
 
 TEST(authenticate, bad_cert_zip) {
-  boost::filesystem::path filepath = "tests/sota_tools/auth_test_cert_bad.zip";
+  // Tries to authenticates with ssl_server on port 1443.
+  // Fails because the intermediate cert that signed the client cert was signed
+  // by a different root cert.
+  boost::filesystem::path filepath = certs_dir / "bad.zip";
+  ServerCredentials creds(filepath);
+  EXPECT_EQ(creds.GetMethod(), AuthMethod::kTls);
   TreehubServer treehub;
-  int r = authenticate("", ServerCredentials(filepath), treehub);
+  int r = authenticate("", creds, treehub);
   EXPECT_EQ(0, r);
   CurlEasyWrapper curl_handle;
   curlEasySetoptWrapper(curl_handle.get(), CURLOPT_VERBOSE, 1);
```

Comment:
Does it make sense to continue the test control flow if the authentication fails, if not then ASSERT_EQ should serve better here.

Reply:
The `EXPECT_EQ` shouldn't hurt and it's not like we'll save a huge amount of test time if we fail earlier. We normally use `EXPECT` for everything unless there's a good reason not to. If there is consensus around changing that norm, I'm open to revising it, though.

Reply:
No worries, I don't think that there is much practical benefits in it.